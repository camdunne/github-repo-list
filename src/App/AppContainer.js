import { connect } from 'react-redux'
import App from './App'

const mapStateToProps = (state) => ({
  apiKey: state.apiKey
});

export default connect(
  mapStateToProps
)(App)