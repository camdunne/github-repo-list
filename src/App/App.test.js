import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { createStore } from 'redux';
import AppContainer from './AppContainer';
import rootReducer from '../reducers';

const store = createStore(rootReducer);

it('renders without crashing', () => {
  const div = document.createElement('div');
  render(
    <Provider store={store}>
      <AppContainer />
    </Provider>,
    div
  );
  unmountComponentAtNode(div);
});

it('App matches snapshot', () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <AppContainer />
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});