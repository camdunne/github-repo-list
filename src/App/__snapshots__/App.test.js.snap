// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App matches snapshot 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <div
      className="EntryForm"
    >
      <form
        action="POST"
        onSubmit={[Function]}
      >
        <label
          htmlFor="apiKey"
        >
          Github API Key:
        </label>
        <input
          name="apiKey"
          type="text"
        />
        <input
          name="Enter"
          type="submit"
        />
      </form>
    </div>
  </header>
</div>
`;

exports[`renders correctly 1`] = `
<div
  className="App"
>
  <header
    className="App-header"
  >
    <div
      className="EntryForm"
    >
      <form
        action="POST"
        onSubmit={[Function]}
      >
        <label
          htmlFor="apiKey"
        >
          Github API Key:
        </label>
        <input
          name="apiKey"
          type="text"
        />
        <input
          name="Enter"
          type="submit"
        />
      </form>
    </div>
  </header>
</div>
`;
